### PR TITLE
BugFix: fixed types to allow for single argument; added useRouteParam to vue-compositions

### PR DESCRIPTION
**Pull request by stackoverfloweth**
fixes this error
image

### CODE REVIEW

**Review by pleek91 — APPROVED** [review]

File: src/useRouteParam/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# useRouteParam
+The `useRouteParam` gives you a readonly reactive value for a query param. If you need the param to be writable, consider [useRouteQueryParam](https://github.com/PrefectHQ/vue-compositions/tree/main/src/useRouteQueryParam).
```
Comment (pleek91) [inline src/useRouteParam/README.md]: It doesn't give you access to query params. It gives you the value of a route param. They're different. 

route.params[param]

Vs

route.query[param]

**Review by pleek91 — CHANGES_REQUESTED** [review]

File: src/useRouteParam/README.md (hunk 1 above)
Comment (stackoverfloweth) [reply src/useRouteParam/README.md]: already fixed

File: src/useRouteParam/useRouteParam.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-redeclare */
+import { computed, Ref } from 'vue'
+import { useRoute } from 'vue-router'
+
+export function useRouteParam(param: string): Ref<string | string[]>
+export function useRouteParam(param: string, defaultValue: string): Ref<string>
+export function useRouteParam(param: string, defaultValue: string[]): Ref<string[]>
+export function useRouteParam(param: string, defaultValue?: string | string[]): Ref<string | string[]> {
```
Comment (pleek91) [inline src/useRouteParam/useRouteParam.ts:8]: This should not support default values. They either exist in the url or they don't.

**Review by pleek91 — CHANGES_REQUESTED** [review]

File: src/useRouteParam/useRouteParam.ts (hunk 2 above)
Comment (pleek91) [reply src/useRouteParam/useRouteParam.ts]: I wonder about the return type as well. Technically route params can be `string` or `string[]`. So technically this is correct. But I believe we've been using this assuming it's `string` and we'd have to do an additional type check every time we use this. 

I'd like to see a method for getting a param that returns `string` so we don't have to do additional type guarding. (maybe an additional method, or a `first` optional Boolean argument?)

**Review by pleek91 — APPROVED** [review]

**Pull request merged**